src/dom-shim.js

```diff
@@ -1,3 +1,57 @@
+/* eslint-disable no-warning-comments */
+
+import { parseFragment, serialize } from 'parse5';
+
+// TODO Should go into utils file?
+function isShadowRoot(element) {
+  return Object.getPrototypeOf(element).constructor.name === 'ShadowRoot';
+}
+
+// Deep clone for cloneNode(deep) - TODO should this go into a utils file?
+// structuredClone doesn't work with functions. TODO This works with
```

Comment:
Are you suggesting a library would / could handle functions?  If we think that feature would be helpful, I would prefer we just extend this to support our needs if so.  (would prefer to keep dependencies in this project as slim as possible and already watch to try and better de-couple the JSX ones).

So yeah, if we want to keep iterating on this, I am happy to have an issue / PR to do, and we can clean up the TODO comment.